src/hpr_config_worker.erl

```diff
@@ -0,0 +1,165 @@
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+-spec start_link(config_worker_opts()) -> any().
+start_link(#{host := Host, port := Port} = Args) when is_list(Host) andalso is_number(Port) ->
+    gen_server:start_link({local, ?SERVER}, ?SERVER, Args, []);
+start_link(#{host := Host, port := PortStr} = Args) when is_list(Host) andalso is_list(PortStr) ->
+    gen_server:start_link(
+        {local, ?SERVER}, ?SERVER, Args#{port := erlang:list_to_integer(PortStr)}, []
+    );
+start_link(_) ->
+    ignore.
+
+%% ------------------------------------------------------------------
+%% gen_server Function Definitions
+%% ------------------------------------------------------------------
+
+init(#{host := Host, port := Port} = Args) ->
+    Path = maps:get(file_backup_path, Args, undefined),
+    State = #state{
+        host = Host,
+        port = Port,
+        connection = undefined,
+        stream = undefined,
+        file_backup_path = Path
+    },
+    lager:info("starting config worker ~s:~w file=~s", [Host, Port, Path]),
+    ok = maybe_init_from_file(State),
+    {ok, State, {continue, ?CONNECT}}.
+
+handle_continue(?CONNECT, #state{host = Host, port = Port} = State) ->
+    case grpc_client:connect(tcp, Host, Port) of
+        {ok, Connection} ->
+            lager:info("connected"),
+            {noreply, State#state{connection = Connection}, {continue, ?INIT_STREAM}};
+        {error, _E} ->
+            lager:error("failed to connect ~p", [_E]),
+            timer:sleep(timer:seconds(1)),
+            {noreply, State, {continue, ?CONNECT}}
+    end;
+handle_continue(
+    ?INIT_STREAM,
+    #state{
+        connection = Connection
+    } = State
+) ->
+    {ok, Stream} = grpc_client:new_stream(
+        Connection, 'helium.config.config_service', route_updates, client_config_pb
+    ),
+    %% Sending Route Request
+    RouteReq = #{},
+    ok = grpc_client:send(Stream, RouteReq),
+    lager:info("stream initialized"),
+    {noreply, State#state{stream = Stream}, {continue, ?RCV_CFG_UPDATE}};
+handle_continue(?RCV_CFG_UPDATE, #state{connection = Connection, stream = Stream} = State) ->
+    case grpc_client:rcv(Stream, ?RCV_TIMEOUT) of
+        {headers, _Headers} ->
+            {noreply, State, {continue, ?RCV_CFG_UPDATE}};
+        {data, RoutesResV1} ->
+            lager:info("got router update"),
+            ok = process_routes_update(RoutesResV1, State),
+            {noreply, State, {continue, ?RCV_CFG_UPDATE}};
+        eof ->
+            lager:warning("got eof"),
+            _ = grpc_client:stop_connection(Connection),
+            {noreply, State, {continue, ?CONNECT}};
+        {error, timeout} ->
+            {noreply, State, {continue, ?RCV_CFG_UPDATE}};
+        {error, E} ->
+            lager:error("failed to rcv ~p", [E]),
+            {stop, {error, E}}
```

Comment:
The main supervisor is a `rest_for_one`, and this worker is second.
If we go down when this connection closes, we're going to close _all_ connections.

Same with the `stop`s in `handle_call` and `handle_cast`.

Reply:
Should the main supervisor be `one_for_one`?